### DISCUSSION TITLE
Alternative to SDParameterGenerator or optimization switch

### DISCUSSION THREAD

**Discussion by fernicar** [receyuki/comfyui-prompt-reader-node#84] [General]
In SDParameterGenerator, the simple change of any value will make the node use more time than vanilla nodes, I think the purpose of SDParameterGenerator is to avoid creating each vanilla node required by SDPromptSaver, In other words, If I don't want to invest time creating many nodes I could use SDParameterGenerator, however, as of now, it is slower than the alternative.

There may be some aproach to this issue, each one with pros and cons.

1. Output combo type of the model/vae name without loading the model in the node itself, A custom node name Easy PromptLine does return a combo, it can be filled by any combo, ie. model names, here is the link of the return type I'm talking about: https://github.com/yolain/ComfyUI-Easy-Use/blob/686aef440915ab6aff556d1122bcb846cd71c8fc/py/easyNodes.py#L311
2. Alternatively you could add 2 new nodes meant to replace SDParameterGenerator, I assume the con would be more nodes to mantain, one of them will be the exact same as the loader from comfyui, however, all the dynamic values will not take as much time as the current SDParameterGenerator on each change, since the model would remain in the second node anyway.
3. Find a way to cache the model loaded by SDParameterGenerator, I don't know if this is possible, then the switch could "lock" the model.
4. Alternative appoach of getting the parameters needed by SDPromptSaver in the node itself, the con may be that it is somewhat more complicated, here is a different approach to get each values without nodes: https://github.com/audioscavenger/save-image-extended-comfyui

A penny for your thoughts?
Cheers

**Comment by receyuki** [comment]
Thanks a lot for your suggestion! I've been super busy lately, so I don't have time to maintain this repo at the moment. I probably won't be making a decision in the next few weeks. Once I have some free time, I'll go through each aproach you've mentioned.
